### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by tosan88**
Changes include:

* Handling some errors during consuming messages as fatal errors
* Improved logging
* Improved formatting
* Other small improvements

NOTE: 
  * ~~Some tests are failing~~
  * ~~no tests for the new feature~~
  * also depending on underlying lib's PR: https://github.com/Financial-Times/kafka-client-go/pull/14

### CODE REVIEW

File: dispatch/dispatcher.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package dispatcher
+package dispatch
```
Comment (tamas-molnar) [inline dispatch/dispatcher.go]: Why this change now? Makes the diff very big.

Comment (tosan88) [reply dispatch/dispatcher.go]: Yeah, maybe I should have left it out

**Comment by tosan88** [comment]
:+1: for changes by @tamas-molnar

File: resources/validator.go [hunk 2]
```diff
@@ -7,29 +7,31 @@ import (
...
 
 	req.Header.Set(apiKeyHeaderField, providedApiKey)
 
-	//if the api key has more than four characters we want to log the first four
+	//if the api key has more than 8 characters we want to log the first and last four
 	apiKeyFirstChars := ""
-	if len(providedApiKey) > 4 {
-		apiKeyFirstChars = providedApiKey[0:4]
+	apiKeyLastChars := ""
+	if len(providedApiKey) > 8 {
```
Comment (peteclark-ft) [inline resources/validator.go]: What's the value in logging the last 4 digits of the API key?

File: resources/validator.go [hunk 3]
```diff
@@ -7,29 +7,31 @@ import (
...
 	if err != nil {
-		log.WithField("url", masheryApiKeyValidationURL).WithError(err).Error("Invalid URL for api key validation")
+		log.WithField("url", apiGatewayApiKeyValidationURL).WithError(err).Error("Invalid URL for api key validation")
 		return false, "Invalid URL", http.StatusInternalServerError
 	}
 
 	req.Header.Set(apiKeyHeaderField, providedApiKey)
 
-	//if the api key has more than four characters we want to log the first four
+	//if the api key has more than 8 characters we want to log the first and last four
```
Comment (peteclark-ft) [inline resources/validator.go]: What is the standard length for tyk API keys? I'm just not sure this is overly secure.. What if there are 5 or 9 chars in the api key? An attacker with this log line would then have a ~1/62 chance of correctly guessing the remaining character.

Should this be based on a percentage of the length of the key, capped at 4 chars?

File: supervisor_test.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+			atomic.AddInt32(&handlerCalls, 1)
+		}
+
+		supervisor := newServiceSupervisor(serviceName, errCh, fatalErrs, fatalErrHandler)
+		go supervisor.Supervise()
+		for _, err := range tc.inputErrs {
+			errCh <- err
+		}
+
+		time.Sleep(250 * time.Millisecond)
```
Comment (peteclark-ft) [inline supervisor_test.go]: V Minor, but you could probably speed up this test by using a WaitGroup in the handler or something similar

File: resources/validator.go [hunk 5]
```diff
@@ -43,10 +45,10 @@ func isValidApiKey(providedApiKey string, masheryApiKeyValidationURL string, htt
...
 
 	if respStatusCode == http.StatusUnauthorized {
-		log.WithField("apiKeyFirstChars", apiKeyFirstChars).Error("Invalid api key")
+		log.WithField("apiKeyFirstChars", apiKeyFirstChars).WithField("apiKeyLastChars", apiKeyLastChars).Error("Invalid api key")
 		return false, "Invalid api key", http.StatusUnauthorized
 	}
 
-	log.WithField("url", req.URL.String()).Errorf("Received unexpected status code from Mashery: %d", respStatusCode)
+	log.WithField("url", req.URL.String()).WithField("apiKeyFirstChars", apiKeyFirstChars).WithField("apiKeyLastChars", apiKeyLastChars).Errorf("Received unexpected status code from Api Gateway: %d", respStatusCode)
 	return false, "Request to validate api key returned an unexpected response", http.StatusServiceUnavailable
```
Comment (peteclark-ft) [inline resources/validator.go]: Should we not also be forwarding `429` and `403` errors to the client here? As we know, these are legitimate responses from Tyk for overuse of API keys.

File: resources/validator.go (hunk 3 above)
Comment (tamas-molnar) [reply resources/validator.go]: The gateway team recommend we log the last 5 chars of a key, since the new Tyk keys have identical prefixes. This will be fixed here https://github.com/Financial-Times/notifications-push/compare/key-validation-fixes?expand=1

File: resources/validator.go (hunk 5 above)
Comment (tamas-molnar) [reply resources/validator.go]: Yes we should, that will be fixed in https://github.com/Financial-Times/notifications-push/compare/key-validation-fixes?expand=1
